# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/ScriptLoader.h
# Commit: 9276c7e1ddd9
# Full Hash: 9276c7e1ddd98ccc725dbbe43353a8d988f8f454
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-12-15 09:27:59
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement initial Static Module Loading for Workers; r=jonco
#   
#   This is the first pass of getting static module loading to work. This roughly implements
#   https://html.spec.whatwg.org/multipage/webappapis.html#fetch-a-worklet/module-worker-script-graph --
#   without some of the settings objects correctly set.
# ==============================================================================

diff -r 6ee318df6641 -r 9276c7e1ddd9 dom/workers/ScriptLoader.h
--- a/dom/workers/ScriptLoader.h	Wed Dec 14 14:55:35 2022 +0000
+++ b/dom/workers/ScriptLoader.h	Wed Dec 14 14:55:35 2022 +0000
@@ -8,6 +8,7 @@
 #define mozilla_dom_workers_scriptloader_h__
 
 #include "js/loader/ScriptLoadRequest.h"
+#include "js/loader/ModuleLoadRequest.h"
 #include "js/loader/ModuleLoaderBase.h"
 #include "mozilla/dom/WorkerCommon.h"
 #include "mozilla/dom/WorkerLoadContext.h"
@@ -142,6 +143,7 @@
   friend class CacheLoadHandler;
   friend class CacheCreator;
   friend class NetworkLoadHandler;
+  friend class WorkerModuleLoader;
 
   RefPtr<ThreadSafeWorkerRef> mWorkerRef;
   UniquePtr<SerializedStackHolder> mOriginStack;
@@ -189,6 +191,8 @@
                             const mozilla::Encoding* aDocumentEncoding,
                             bool aIsMainScript);
 
+  ScriptLoadRequest* GetMainScript();
+
   already_AddRefed<ScriptLoadRequest> CreateScriptLoadRequest(
       const nsString& aScriptURL, const mozilla::Encoding* aDocumentEncoding,
       bool aIsMainScript);
@@ -202,6 +206,8 @@
 
   nsIURI* GetInitialBaseURI();
 
+  nsIGlobalObject* GetGlobal();
+
   void MaybeMoveToLoadedList(ScriptLoadRequest* aRequest);
 
   bool StoreCSP();
@@ -239,8 +245,6 @@
 
   nsTArray<RefPtr<ThreadSafeRequestHandle>> GetLoadingList();
 
-  nsIGlobalObject* GetGlobal();
-
   bool EvaluateScript(JSContext* aCx, ScriptLoadRequest* aRequest);
 
   nsresult FillCompileOptionsForRequest(